README.md

```diff
@@ -27,11 +27,32 @@ For more complex hierarchies, the `values` list can be a well-defined order base
 
 Creating these lists manually [is possible](https://discuss.kotlinlang.org/t/list-of-sealed-class-objects), but maintaining them is error prone and doesn't solve the problem generically.
 
-This annotation processor automatically creates and maintains those lists to allow sealed classes of only objects (sealed enums, if you will) to be strictly more feature-rich than normal enums.
+In addition, unique names can be associated with these objects, and thus can be converted to and from a string representation.
+
+Thus, we can define an interface `SealedEnum` that, when implemented, provides equivalent functionality to a normal enum, generically:
+
+```kotlin
+interface SealedEnum<T> : Comparator<T> {
+
+    val values: List<T>
```

Comment:
It _is_ possible, but slightly clunky.

What we could do is have

```kotlin
val MyPet.Companion.values: List<TwoObjectSealedClass> get() = ...
```

However, this requires that `MyPet` has a `companion object` defined, since we can't edit the code as an annotation processor:

```kotlin
@GenSealedEnum
sealed class MyPet {
  object Fido : MyPet()
  object Cleo : MyPet()

  companion object
}
```

It wouldn't be inconceivable to require that the user define the unnamed `companion object`; Arrow, for instance, does this for some of its processors.

Reply:
Ha, your edit beat me to it for mentioning Arrow!

The downside with removing the `SealedEnum` object completely is that it makes generics hard to work with, since we can't make the `companion object` implement an interface just with extensions. We could do some sort of mix, in addition to having

```kotlin
val MyPet.Companion.sealedEnum: SealedEnum<MyPet> get() = MyPetSealedEnum
```

Reply:
I was already thinking that it'd be nice to generate all of those extensions functions to make the consumer side closer to a normal enum:

```
val MyPet.ordinal: Int get() = MyPetSealedEnum.ordinalOf(this)

val MyPet.name: String get() = MyPetSealedEnum.nameOf(this)
```

Reply:
That would keep the flexibility of the full object for generic purposes, while also being the easiest to use in the common case of not having to deal with generics.

Reply:
For completeness, this would be all of the extension functions we could add:
- `val MyPet.ordinal: Int`
- `val MyPet.name: String`
- If the companion object exists: **Q <-- Do we require this?** 
  - `val MyPet.Companion.values: List<MyPet>` / `fun MyPet.Companion.values(): List<MyPet>` **Q <-- Is `val`/`fun` better or more consistent here?**
  - `val MyPet.Companion.sealedEnum: SealedEnum<MyPet>`
  - `val MyPet.Companion.valueOf(name: String): MyPet`
- If the enum class is generated
  - `val MyPet.enum: MyPetEnum`
  - `val MyPetEnum.sealedObject: MyPet`

Especially if we require that the companion object exists, a nice bonus of all of these would be that it'd be very unlikely to have to refer to the generated classes directly, since everything could be accessed via extension methods.

Reply:
Yeah I really like this.  This pattern would be highly discoverable and feel integrated.  I'm in favor if it as a starting point.

> Especially if we require that the companion object exists, a nice bonus of all of these would be that it'd be very unlikely to have to refer to the generated classes directly, since everything could be accessed via extension methods.

And we can let the 80% bulk of common use cases guide the API from there, whatever they turn out to be.

Reply:
Hmm, okay thinking about this more I found a sticking point: Currently, it's possible to have multiple `SealedEnum`s generated with different traversal orders, which makes the name `MyPet.ordinal` a bit ambiguous.

My plan is to continue to use the traversal order prefix to disambiguate them for the case where there is more than one traversal order defined. That will keep the nice simple names discussed above for the most common case, while still allow complex ones (with more complex names)

Reply:
The traversal order prefix worked for all of the cases, and meshes nicely with the existing names.